libsmartcols/src/filter-param.c

```diff
@@ -523,6 +517,78 @@ static int float_opers(enum filter_etype oper, struct filter_param *l,
 	return 0;
 }
 
+static bool maybe_percentage_opers(enum filter_etype oper, struct filter_param *l,
+				   struct filter_param *r, int *status)
+{
+	long double lfnum, rfnum;
+	struct filter_param lf, rf;
+	int rc;
+
+	if (!is_percentage(l, &lfnum))
+		return false;
+
+	if (!is_percentage(r, &rfnum))
+		return false;
+
+	lf = *l;
+	lf.type = SCOLS_DATA_FLOAT;
+	if (!lf.empty)
+		lf.val.fnum = lfnum;
+
+	rf = *r;
+	rf.type = SCOLS_DATA_FLOAT;
+	if (!rf.empty)
+		rf.val.fnum = rfnum;
+
+	rc = float_opers(oper, &lf, &rf, status);
+	if (rc < 0)
+		return false;
+
+	return true;
+}
+
```

Comment:
I'm not sure about this. 

You're trying to detect percentage usage during filter evaluation, but it's obvious already during filter parse time. I'll add separate comment about it.